crates/extension/src/extension_builder.rs

```diff
@@ -371,7 +373,7 @@ impl ExtensionBuilder {
 
         let wasi_sdk_dir = self.cache_dir.join("wasi-sdk");
         let mut clang_path = wasi_sdk_dir.clone();
-        clang_path.extend(["bin", "clang-17"]);
+        clang_path.extend(["bin", &format!("clang{}", env::consts::EXE_SUFFIX)]);
```

Comment:
on windows there's no symlinked `clang-17` binary. I think just using `clang` here on macos/linux is fine? Let me know if this is unacceptable (or the CI will tell me :P)

Reply:
I tested this out locally and it seemed to work fine on macOS.